src/client.ts

```diff
@@ -364,7 +360,13 @@ export class Client extends EventEmitter {
       soapAction = ((ns.lastIndexOf('/') !== ns.length - 1) ? ns + '/' : ns) + name;
     }
 
-    if (!this.wsdl.options.forceSoap12Headers) {
+    if (this.wsdl.options.forceSoap12Headers) {
+      headers['Content-Type'] = 'application/soap+xml; charset=utf-8';
+      if (this.wsdl.options.addHeadersAction) {
```

Comment:
According to specification from https://docs.intersystems.com/latest/csp/docbook/DocBook.UI.Page.cls?KEY=ROBJ_method_soapaction action param should be added within forceSoap12Headers flag, not some additional one